### PR TITLE
Zmtelemetry improvements

### PR DESCRIPTION
**Pull request by connortechnology**

### CODE REVIEW

**Comment by knnniggett** [comment]
zmdc already logs an event when each deamon is stopped or started, including zmtelemetry. So by changing the print statements to INFO, you are now creating two log entries reporting the status of zmtelemetry and nothing gets reported to stdout, should the end user decide to call this from the command line. I don't agree with this.

**Comment by knnniggett** [comment]
This line here is what currently creates a log entry when any daemon is started or stopped:
https://github.com/ZoneMinder/ZoneMinder/blob/master/scripts/zmdc.pl.in#L457

**Comment by connortechnology** [comment]
Hmm.. perhaps they should be debug lines then.

**Comment by knnniggett** [comment]
I don't agree. If I call zmtelemetry.pl from the command line, I want to know it is running, and not locked up, by its output to stdout. In that case, I don't want to have to check a log file. This is how all the other zm scripts work as well, including zmupdate.pl which shares the same lines of code.

It's working fine as it is. Lets focus our limited resources on priorities that are work our while.

File: scripts/ZoneMinder/lib/ZoneMinder/ConfigData.pm.in [hunk 1]
```diff
@@ -2813,6 +2813,23 @@ our @options = (
...
+    category    =>  'system',
+  },
+  {
+    name        => 'ZM_TELEMETRY_SERVER_ENDPOINT',
+    default     => 'https://zmanon:2b2d0b4skps@telemetry.zoneminder.com/zmtelemetry/testing5',
+    description => 'URL that ZoneMinder will send usage data to',
+    help        => qqq("
+        "),
+    type        => $types{url},
+    category    => 'system',
```
Comment (kylejohnson) [inline scripts/ZoneMinder/lib/ZoneMinder/ConfigData.pm.in]: Hmm.  I'm torn on this.  On one hand, it is nice to be able to update the value without modifying the perl script.  But that itself isn't really a pain point.

My real issue is with the 'system' category.  This isn't something that the user should be able to (easily) change.  Perhaps the 'hidden' category works better.  Makes it more easily configurable, but doesn't give the user easy access to update it.

File: scripts/zmtelemetry.pl.in [hunk 2]
```diff
@@ -340,4 +356,5 @@ sub linuxDistro {
 return ($kernel, $distro, $version);
 }
 
-
+1;
```
Comment (kylejohnson) [inline scripts/zmtelemetry.pl.in:347]: What does this do?

File: scripts/zmtelemetry.pl.in [hunk 3]
```diff
@@ -340,4 +356,5 @@ sub linuxDistro {
...
+__END__
```
Comment (kylejohnson) [inline scripts/zmtelemetry.pl.in:348]: And this one?

Comment (connortechnology) [reply scripts/zmtelemetry.pl.in:348]: The 1 gives a defined exit value.  The __END__ tells perl that no further processing should be done.  So you can put the documentation after this line, using perldoc, etc and it won't be executed.  This is the standard, preferred thing to put at the end of a perl script.

File: scripts/ZoneMinder/lib/ZoneMinder/ConfigData.pm.in (hunk 1 above)
Comment (connortechnology) [reply scripts/ZoneMinder/lib/ZoneMinder/ConfigData.pm.in]: I'm fine with putting it in hidden.

**Comment by connortechnology** [comment]
in regards to the print/Info thing... I'm doing some experimenting, trying to figure out what might be the best.
The thing is, when I run it, it says it is starting up.... and then nothing.  So I need it to output MORE.  In fact, I'm tempted to want it to output debug level statements, because if I am running it from the command line, I am probably doing so in an effort to debug what it is doing.  

So in my testing, I added a line to the logger module that tests for stdout and if so defaults the termLevel to Info, which makes it output Info level logs to the terminal.  This works well although it is output log lines which is a bit icky in terms of formatting.... So I'm not sure this is the way to go... but I kinda like it.  

ANyways, I'm not terribly opposed to putting those lines back to print's, but I want to think about it a bit first.

**Comment by connortechnology** [comment]
For the sake of getting this done, I've reverted the print/Info stuff.  And I've hidden the ENDPOINT.

Further thoughts:  Why are we sleeping an hour?  How about we sleep $now-$lastcheck + 1 hour?

Copyright thoughts: all our copyrights assign to Phil.  Perhaps we should now assign them to ZoneMinder LLC?

**Comment by kylejohnson** [comment]
> Copyright thoughts: all our copyrights assign to Phil. Perhaps we should now assign them to ZoneMinder LLC?

Yeah, there's a lot of code that needs this updated.  Should definitely come in a separate PR after some discussion.

**Comment by kylejohnson** [comment]
Looks OK to me.  @knnniggett ?

**Comment by knnniggett** [comment]
We can do a search and replace on the license text same way I recently updated the gpl license mailing address.

@connortechnology Why did you move the pod text to the bottom in this PR when all the other perl scripts have the pod text at the top? I personally don't care one way or the other, but I do want consistency.

**Comment by connortechnology** [comment]
because the perl parser stops after the __END__ so it is more efficient to put it there. I agree about consistency, but all the other scripts should be updated as well.

**Comment by knnniggett** [comment]
That's exactly the point ...go update all the other scripts to be consistent, or move the change you've got in this pr into a new pr. You're asking us to merge a PR with known, unfinished changes. Surely you can understand why I won't merge this as it is, right?

**Comment by connortechnology** [comment]
There is very ilttle consistency in the perl scripts, and the case of the pod documentation being at the end is true of every single perl module that we provide.  So I am bringing zmtelemetry into compliance.  

I will be happy to make another pr that brings the other scripts into compliance, but that is not appropriate for this PR.

**Comment by kylejohnson** [comment]
I get the feeling that Andy wants the pod text change reverted as opposed to an ad-hoc change here?
With that said, how about a New Year's Resolution for the team: focus on what matters.  Changing the pod text location, and even the existing GPL text, doesn't get us anything and certainly doesn't matter right now.

**Comment by connortechnology** [comment]
I'm so tired of working on ZM.  I think I'll step away for a bit.

**Pull request closed without merging**